### DISCUSSION TITLE
"routingKey" pattern

### DISCUSSION THREAD

**Discussion by AndreYurin** [golevelup/nestjs#629] [Q&A]
Hello!
First of all, thanks for the lib. So far everything is working well!

I have a question about the RabbitMQ lib:
Could you please answer if it supports "message pattern" with * or # (https://www.rabbitmq.com/tutorials/tutorial-five-javascript.html)? The binding works, I can see it in RabbitMQ, but the consumer does not receive messages. It receives message only if routingKey of message will be exactly with * or #. For example:
I have exchange "test" with "topic" type.

I use
`@RabbitRPC({
        exchange: 'test',
        queue: 'testqueue',
        routingKey: 'testing.#'
})`

I publish message with 
`const response = await this.amqpConnection.request({
      exchange: 'test',
      routingKey: 'testing.test',
      payload: {
        message: "value",
      }
 });`
but consumer doesn't receive this message. 

It will get it if I publish with
`const response = await this.amqpConnection.request({
      exchange: 'test',
      routingKey: 'testing.#',
      payload: {
        message: "value",
      }
 });`

**Comment by amroczeK** [comment]
You are using the incorrect decorator for a Topic Exchange. Use RabbitSubscribe instead.

```
@RabbitSubscribe({
    exchange: 'test',
    routingKey: 'testing.#',
    queue: 'testqueue',
  })
```

Here is another example of how I am using it:
```
@RabbitSubscribe({
    exchange: 'shop.topic',
    routingKey: 'shop.inventory.*',
    queue: 'inventory',
  })
  async handleInventory(
    @RabbitPayload()
    inventoryData: any,
    @RabbitRequest() request: IRabbitRequest,
  ): Promise<void> {
    this.logger.log(`handleInventory()`);
    const routingKey = request.fields.routingKey;
    switch (routingKey) {
      case 'shop.inventory.create': {
        // TODO: Validate payload
        this.inventoryService.createInventory(inventoryData);
        break;
      }
      case 'shop.inventory.list': {
        // TODO: Validate payload
        this.inventoryService.listInventory();
        break;
      }
      case 'shop.inventory.update': {
        // TODO: Validate payload
        this.inventoryService.updateInventory(inventoryData);
        break;
      }
      case 'shop.inventory.delete': {
        // TODO: Validate payload
        this.inventoryService.removeFromInventory(inventoryData);
        break;
      }
      case 'shop.inventory.update.quantity': {
        // TODO: Validate payload
        this.inventoryService.updateInventory(inventoryData);
        break;
      }
      default: {
        this.logger.log(
          `There is no handler for message with routing key: ${routingKey}`,
        );
        break;
      }
    }
  }
```
